quine/src/main/scala/com/thatdot/quine/app/importers/WebsocketSimpleStartup.scala

```diff
@@ -0,0 +1,131 @@
+package com.thatdot.quine.app.importers
+
+import java.nio.charset.Charset
+
+import scala.compat.ExecutionContexts
+import scala.concurrent.duration.DurationInt
+import scala.util.{Failure, Success}
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.IllegalUriException
+import akka.http.scaladsl.model.ws.{
+  BinaryMessage,
+  InvalidUpgradeResponse,
+  Message,
+  TextMessage,
+  ValidUpgrade,
+  WebSocketRequest
+}
+import akka.http.scaladsl.settings.ClientConnectionSettings
+import akka.stream.KillSwitches
+import akka.stream.contrib.{SwitchMode, Valve}
+import akka.stream.scaladsl.{Flow, Keep, Source}
+import akka.util.ByteString
+
+import com.typesafe.scalalogging.LazyLogging
+
+import com.thatdot.quine.app.ControlSwitches
+import com.thatdot.quine.app.importers.WebsocketSimpleStartup.UpgradeFailedException
+import com.thatdot.quine.app.importers.serialization.ImportFormat
+import com.thatdot.quine.app.routes.IngestMeter
+import com.thatdot.quine.graph.CypherOpsGraph
+import com.thatdot.quine.graph.MasterStream.{IngestSrcExecToken, IngestSrcType}
+import com.thatdot.quine.routes.WebsocketSimpleStartupIngest
+import com.thatdot.quine.routes.WebsocketSimpleStartupIngest.KeepaliveProtocol
+object WebsocketSimpleStartup {
+  class UpgradeFailedException(cause: Throwable)
+      extends RuntimeException("Unable to upgrade to websocket connection", cause) {
+    def this(cause: String) = this(new Throwable(cause))
+  }
+}
+case class WebsocketSimpleStartup(
+  format: ImportFormat,
+  wsUrl: String,
+  initMessages: Seq[String],
+  keepaliveProtocol: KeepaliveProtocol,
+  parallelism: Int,
+  charset: Charset,
+  charsetTranscoder: Flow[ByteString, ByteString, NotUsed],
+  meter: IngestMeter,
+  initialSwitchMode: SwitchMode
+) extends LazyLogging {
+  @throws[IllegalUriException]("When the provided url is invalid")
```

Comment:
Is this `@throws` accurate? I can only trigger `IllegalArgumentException` from `akka-http` https://github.com/akka/akka-http/blob/7476d36aa7985d37469551b54066074370ce5d61/akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala#L704-L706. (Obviously I can also get `UpgradeFailedException`, but those just fail the stream - they aren't thrown in the method.)